### ISSUE TITLE
[Elastica] Processors are not registered

### ISSUE THREAD

**Issue by philmoreau** [php-enqueue/enqueue-dev#812]
I've been following the documentation about [populate](https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/doc/cookbook/speed-up-populate-command.md) and [Doctrine](https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/doc/cookbook/doctrine-queue-listener.md). Unfortunately the examples do not seem to produce de desired results. 

The command "bin/console enqueue:transport:consume enqueue_elastica.populate_processor -vvv", leads to :

```
In ContainerProcessorRegistry.php line 24:
                                                                                              
  [LogicException]                                                                            
  Service locator does not have a processor with name "enqueue_elastica.populate_processor".

```

The command "bin/console enqueue:transport:consume enqueue_elastica.doctrine.sync_index_with_object_change_processor -vvv", leads to :

```
In ContainerProcessorRegistry.php line 24:
                                                                                                                            
  [LogicException]                                                                                                          
  Service locator does not have a processor with name "enqueue_elastica.doctrine.sync_index_with_object_change_processor".

```

Debugging the routes with "enqueue:routes" leads to `Found 0 routes`.

However the processors are registered as services :

```
Information for Service "enqueue_elastica.doctrine.sync_index_with_object_change_processor"
===========================================================================================

 ---------------- -------------------------------------------------------------------------- 
  Option           Value                                                                     
 ---------------- -------------------------------------------------------------------------- 
  Service ID       enqueue_elastica.doctrine.sync_index_with_object_change_processor         
  Class            Enqueue\ElasticaBundle\Doctrine\Queue\SyncIndexWithObjectChangeProcessor  
  Tags             enqueue.command_subscriber (client: fs)                                   
                   enqueue.transport.processor (transport: fs)                               
  Public           no                                                                        
  Synthetic        no                                                                        
  Lazy             no                                                                        
  Shared           yes                                                                       
  Abstract         no                                                                        
  Autowired        no                                                                        
  Autoconfigured   no                                                                        
 ---------------- -------------------------------------------------------------------------- 

```

Can anyone else reproduce this issue?

**Comment by philmoreau** [comment]
Downgrading to version 0.8.x of "enqueue/elastica-bundle", "enqueue/enqueue" and "enqueue/fs" fixes the problem.

**Comment by makasim** [comment]
Is the issue still there? Have you found anything?

**Comment by philmoreau** [comment]
I thought it was related to #789 but using @dkarlovi 's [solution](https://github.com/php-enqueue/enqueue-dev/issues/789#issuecomment-468174256) did not solve the problem, so I downgraded to 0.8 and kept it.

Sorry, not much help I know but that is all I can tell you right now.

**Comment by KonstantinCodes** [comment]
I have the same problem.

**Comment by makasim** [comment]
@KonstantinCodes PR is very welcome.

**Comment by KonstantinCodes** [comment]
@makasim We may just not have configured the bundle correctly. As I understand, the processor just needs the tag `enqueue.topic_subscriber` and then should be automatically registered for the `default` client. But when I receive a message, enqueue always calls the DelegateProcessor instead of my Processor. Then it fails because the messages which I receive don't contain Processor metadata, so the exception is thrown.

I didn't find the solution yesterday. I can imagine that I just didn't configure my Processor correctly. Then again, I did it exactly as in the docs. https://php-enqueue.github.io/bundle/quick_tour/

**Comment by KonstantinCodes** [comment]
The only solution we came up with is to make an Extension, which sets the custom Processor on the actual Message, so that `\Enqueue\Client\DelegateProcessor` will call it.
It does work, but I'm not sure, this is the preferred way of implementing the Custom Processor.

Should I make a PR, documenting the Extension in the link I provided in the last comment?
Or is this the wrong approach in your opinion?
